$OpenBSD: patch-error_c,v 1.2 2012/10/08 13:15:17 jeremy Exp $

Backport security fix, from Shugo Maeda (ruby developer), similar to
changes in ruby SVN revision 37068.

--- error.c.orig	Mon Oct  8 03:43:34 2012
+++ error.c	Mon Oct  8 03:43:49 2012
@@ -665,9 +665,6 @@ name_err_to_s(exc)
 
     if (NIL_P(mesg)) return rb_class_name(CLASS_OF(exc));
     StringValue(str);
-    if (str != mesg) {
-	OBJ_INFECT(str, mesg);
-    }
     return str;
 }
 
